connectors/sources/redis.py

```diff
@@ -0,0 +1,283 @@
+            _type: Datatype for filter data
+
+        Yields:
+            list: list of keys in a database
+        """
+        cursor, keys = await redis_client.scan(
+            match=pattern, count=PAGE_SIZE, _type=_type
+        )
+        yield keys
+
+        while cursor != 0:
+            cursor, keys = await redis_client.scan(
+                cursor=cursor, match=pattern, count=PAGE_SIZE
+            )
+            yield keys
+
+    async def get_key_value(self, redis_client, key, key_type):
+        """Fetch value of key for database.
+
+        Args:
+            redis_client: redis client object
+            key: Name of key
+            key_type: Type of key
+
+        Yields:
+            Value of key
+        """
+        try:
+            if key_type == KeyType.HASH.value:
+                return await redis_client.hgetall(key)
+            elif key_type == KeyType.STREAM.value:
+                return await redis_client.xread({key: "0"})
+            elif key_type == KeyType.LIST.value:
+                return await redis_client.lrange(key, 0, -1)
+            elif key_type == KeyType.SET.value:
+                return await redis_client.smembers(key)
+            elif key_type == KeyType.ZSET.value:
+                return await redis_client.zrange(key, 0, -1, withscores=True)
+            elif key_type == KeyType.JSON.value:
+                value = await redis_client.execute_command("JSON.GET", key)
+                return json.loads(value)
+            else:
+                value = await redis_client.get(key)
+                if value is not None:
+                    try:
+                        value_json = json.loads(value)
+                        return value_json
+                    except json.JSONDecodeError:
+                        return value
+        except UnicodeDecodeError as exception:
+            self._logger.warning(
+                f"Something went wrong while fetching value for key {key}. Error: {exception}",
+                exc_info=True,
+            )
+            return ""
+
+
+class RedisDataSource(BaseDataSource):
+    """Redis"""
+
+    name = "Redis"
+    service_type = "redis"
+
+    def __init__(self, configuration):
+        super().__init__(configuration=configuration)
+        self.redis_client = RedisClient(configuration=configuration)
+
+    @classmethod
+    def get_default_configuration(cls):
+        return {
+            "host": {"label": "Host", "order": 1, "type": "str"},
+            "port": {"label": "Port", "order": 2, "type": "int"},
+            "username": {
+                "label": "Username",
+                "order": 3,
+                "required": False,
+                "type": "str",
+            },
+            "password": {
+                "label": "Password",
+                "order": 4,
+                "required": False,
+                "sensitive": True,
+                "type": "str",
+            },
+            "database": {
```

Comment:
Is this a product ask to have this field?

Reply:
I guess `databases` misunderstood here. Basically we are passing the database number to which user want to connect as for redis server we have default 16 index available that can be changed from redis.conf file.

Reply:
Yes, I am concerned about this because I see the database is just a number. Is it easy for a customer to get this number? It's not as intuitive as `tables` in MySQL data source. Is this a feature expected from customers?

Reply:
In redis database is referred with number only. In below image you can see the edit option where we can change the database number.  For better understanding we can change the name from `databsse--->index`
![image (1)](https://github.com/elastic/connectors/assets/125074214/0ee330c2-5915-4bb6-b40c-2f74e9a6111c)

Reply:
If the term in Redis is called `Database`, then we can leave it as `Database`.